apps/src/lib/node/ledger/shell/prepare_proposal/block_space_alloc.rs

```diff
@@ -0,0 +1,254 @@
+//! Primitives that facilitate keeping track of the number
+//! of bytes utilized by some Tendermint consensus round's proposal.
+//!
+//! This is important, because Tendermint places an upper bound
+//! on the size of a block, rejecting blocks whose size exceeds
+//! the limit stated in [`RequestPrepareProposal`].
+//!
+//! The code in this module doesn't perform any deserializing to
+//! verify if we are, in fact, allocating space for the correct
+//! kind of tx for the current [`BlockSpaceAllocator`] state. It
+//! is up to the user to dispatch the correct kind of tx into the
+//! current state of the allocator.
+//!
+//! # How space is allocated
+//!
+//! In the current implementation, we allocate space for transactions
+//! in the following order of preference:
+//!
+//! - First, we allocate space for DKG decrypted txs.
+//! - Next, we allocate space for protocol txs. Protocol txs get 1/3 of the
+//!   block space allotted to them.
+//! - Finally, we allocate space for DKG encrypted txs.
+//! - If any space remains, we try to fit other smaller txs in the block.
+//!
+//! Since decrypted txs will utilize at most as much space as
+//! encrypted txs will utilize, and we allocate 1/3 of space
```

Comment:
"Since decrypted txs will utilize at most as much space as encrypted txs will utilize" - encrypted txs being included in this block, or encrypted txs from the previous block height (or both)?

Reply:
encrypted txs from the previous block

Reply:
it would be worth being explicit here I think that it is the encrypted txs from the previous block